dlinfer/vendor/maca/csrc/pos_encoding_kernels.cu

```diff
@@ -128,31 +161,34 @@ void rotary_embedding(
     torch::Tensor& key,    // [batch_size, seq_len, num_kv_heads * head_size] or
                            // [num_tokens, num_kv_heads * head_size]
     int64_t head_size,
-    torch::Tensor& cos_sin_cache,  // [max_position, rot_dim]
+    torch::Tensor& cos,  // [max_position, embed_dim]
+    torch::Tensor& sin,  // [max_position, embed_dim]
     bool is_neox) {
+  auto cos_sin_cache = torch::cat({cos, sin}, -1);
+
   int64_t num_tokens = query.numel() / query.size(-1);
-  int rot_dim = cos_sin_cache.size(1);
+  int embed_dim = cos.size(-1);
```

Comment:
what is the shape of cos/sin and cos_sin_cache?

Reply:
cos/sin: [max_position, embed_dim ]
cos_sin_cache:  [max_position, rot_dim]
embed_dim  = rot_dim / 2
only embed_dim  is used in kernel.